pkgs/applications/audio/carla/default.nix

```diff
@@ -1,91 +1,13 @@
-assert withGtk2 -> gtk2 != null;
-assert withGtk3 -> gtk3 != null;
-
-stdenv.mkDerivation rec {
-  pname = "carla";
-  version = "2.5.1";
-
-  src = fetchFromGitHub {
-    owner = "falkTX";
-    repo = pname;
+libsForQt5: {
+  carla = libsForQt5.callPackage ./common.nix rec {
+    version = "2.5.3";
+    hash = "sha256-J0C3GLdlLMkm3LHl6l3OI2rA73A6z5MMcNJ1I1T0pbI=";
     rev = "v${version}";
-    sha256 = "sha256-SN+9Q5v0bv+kQcYLBJmSCd9WIGSeQuOZze8LVwF20EA=";
   };
 
-  nativeBuildInputs = [
-    python3Packages.wrapPython pkg-config which wrapQtAppsHook
-  ];
-
-  pythonPath = with python3Packages; [
-    rdflib pyliblo
-  ] ++ lib.optional withFrontend pyqt5;
-
-  buildInputs = [
-    file liblo alsa-lib fluidsynth jack2 libpulseaudio libsndfile
-  ] ++ lib.optional withQt qtbase
-    ++ lib.optional withGtk2 gtk2
-    ++ lib.optional withGtk3 gtk3;
-
-  propagatedBuildInputs = pythonPath;
-
-  enableParallelBuilding = true;
-
-  installFlags = [ "PREFIX=$(out)" ];
-
-  postPatch = ''
-    # --with-appname="$0" is evaluated with $0=.carla-wrapped instead of carla. Fix that.
-    for file in $(grep -rl -- '--with-appname="$0"'); do
-        filename="$(basename -- "$file")"
-        substituteInPlace "$file" --replace '--with-appname="$0"' "--with-appname=\"$filename\""
-    done
-  '';
-
-  dontWrapQtApps = true;
-  postFixup = ''
-    # Also sets program_PYTHONPATH and program_PATH variables
-    wrapPythonPrograms
-    wrapPythonProgramsIn "$out/share/carla/resources" "$out $pythonPath"
-
-    find "$out/share/carla" -maxdepth 1 -type f -not -name "*.py" -print0 | while read -d "" f; do
-      patchPythonScript "$f"
-    done
-    patchPythonScript "$out/share/carla/carla_settings.py"
-    patchPythonScript "$out/share/carla/carla_database.py"
-
-    for program in $out/bin/*; do
-      wrapQtApp "$program" \
-        --prefix PATH : "$program_PATH:${which}/bin" \
-        --set PYTHONNOUSERSITE true
-    done
-
-    find "$out/share/carla/resources" -maxdepth 1 -type f -not -name "*.py" -print0 | while read -d "" f; do
-      wrapQtApp "$f" \
-        --prefix PATH : "$program_PATH:${which}/bin" \
-        --set PYTHONNOUSERSITE true
-    done
-  '';
-
-  meta = with lib; {
-    homepage = "https://kx.studio/Applications:Carla";
-    description = "An audio plugin host";
-    longDescription = ''
-      It currently supports LADSPA (including LRDF), DSSI, LV2, VST2/3
-      and AU plugin formats, plus GIG, SF2 and SFZ file support.
-      It uses JACK as the default and preferred audio driver but also
-      supports native drivers like ALSA, DirectSound or CoreAudio.
-    '';
-    license = licenses.gpl2Plus;
-    maintainers = [ maintainers.minijackson ];
-    platforms = platforms.linux;
+  carla-unstable = libsForQt5.callPackage ./common.nix {
+    version = "unstable-2023-02-12";
+    hash = "sha256-OegwW8sfUV4IjBFrq38tpKbOMzZT0cvhL/u3Gr1m938=";
+    rev = "4c29367b6e6f82e75746b9ee3a01f9d205c0beae";
```

Comment:
Do we only need this for one package? Then we should not expose it on top-level and rather just do the overwrite for the one package